From 929c83af8d8d1726b45bbcf46482a608b2e9368a Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 10 Mar 2022 16:13:43 +0200
Subject: [PATCH 027/103] arch: arm/mach-s32: Use fdtdec_* to initialize DDR
 banks

Use fdtdec_setup_mem_size_base() and fdtdec_setup_memory_banksize()
functions to initialize DDR banks.

Issue: ALB-8576
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/cpu.c | 33 ++-------------------------------
 arch/arm/mach-s32/s32-cc/soc.c |  3 ++-
 2 files changed, 4 insertions(+), 32 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/cpu.c b/arch/arm/mach-s32/s32-cc/cpu.c
index 5b54178371..c1fe08a343 100644
--- a/arch/arm/mach-s32/s32-cc/cpu.c
+++ b/arch/arm/mach-s32/s32-cc/cpu.c
@@ -14,6 +14,7 @@
 #include <asm-generic/sections.h>
 #include <linux/sizes.h>
 #include <debug_uart.h>
+#include <fdtdec.h>
 
 #define S32GEN1_DRAM_STD_ADDR	0x80000000ULL
 #define S32GEN1_DRAM_EXT_ADDR	0x800000000ULL
@@ -125,39 +126,9 @@ int arch_cpu_init(void)
 	return 0;
 }
 
-static void s32_init_ram_size(void)
-{
-	int i;
-	unsigned long start, size;
-
-	if (!gd->bd) {
-		pr_err("gd->bd isn't initialized\n");
-		return;
-	}
-
-	gd->ram_size = 0;
-	for (i = 0; i < CONFIG_NR_DRAM_BANKS; i++) {
-		start = gd->bd->bi_dram[i].start;
-		size = gd->bd->bi_dram[i].size;
-
-		if (!start && !size)
-			continue;
-
-		gd->ram_size += get_ram_size((long *)start, size);
-	}
-}
-
 int dram_init_banksize(void)
 {
-	int ret;
-
-	ret = fdtdec_setup_memory_banksize();
-	if (ret)
-		return ret;
-
-	s32_init_ram_size();
-
-	return 0;
+	return fdtdec_setup_memory_banksize();
 }
 
 phys_size_t __weak get_effective_memsize(void)
diff --git a/arch/arm/mach-s32/s32-cc/soc.c b/arch/arm/mach-s32/s32-cc/soc.c
index da6c1b11be..9577071b2f 100644
--- a/arch/arm/mach-s32/s32-cc/soc.c
+++ b/arch/arm/mach-s32/s32-cc/soc.c
@@ -10,6 +10,7 @@
 #include <errno.h>
 #include <hang.h>
 #include <board_common.h>
+#include <fdtdec.h>
 #ifdef CONFIG_SAF1508BET_USB_PHY
 #include <dm/device.h>
 #include <dm/device-internal.h>
@@ -21,7 +22,7 @@ DECLARE_GLOBAL_DATA_PTR;
 
 __weak int dram_init(void)
 {
-	return 0;
+	return fdtdec_setup_mem_size_base();
 }
 
 #ifdef CONFIG_SAF1508BET_USB_PHY
-- 
2.25.1

